drizzlepac/haputils/hapcut_utils.py

```diff
@@ -0,0 +1,383 @@
+        return updated_query_table
+
+    # Get the data product list associated with the elements of the table
+    if verbose:
+        print("Get the product list for all entries in the query table.")
+    dp_table = Observations.get_product_list(updated_query_table)
+
+    # Filter on MVM drizzled product
+    suffix = ["DRZ", "DRC"]
+    if verbose:
+        print("Filter the product list table for only {} filenames.".format(suffix))
+    filtered_dp_table = Observations.filter_products(dp_table,
+                                                     productSubGroupDescription=suffix,
+                                                     extension="fits")
+
+    if not filtered_dp_table:
+        print("WARNING: No drizzle product files (DRZ/DRC) found.")
+        return filtered_dp_table
+
+    # Write the filtered data product table out to a file.  This allows for further
+    # manipulation of the information before a list of filenames is distilled from
+    # the table.
+    # Output filename in the form: mvm_query_ra.dddd_sdec.dddd_radius_cutout.ecsv.
+    #                              mvm_query_84.9208_s69.1483_71_cutout.ecsv
+    ns = "s" if sky_coord.dec.degree < 0.0 else "n"
+    query_filename = "mvm_query_" + str(sky_coord.ra.degree) + "_" + ns + str(abs(sky_coord.dec.degree)) + \
+                     "_{:.0f}_cutout".format(radius.value) + ".ecsv" 
+
+    if verbose:
+        print("Writing out the filter product list table to {}.".format(query_filename))
+    filtered_dp_table.write(query_filename, format="ascii.ecsv")
+
+    return filtered_dp_table
+
+
+def mvm_retrieve_files(products, archive=False, clobber=False, verbose=False):
+    """
+    This function retrieves specified files from the archive - unless the file is found
+    to be locally resident on disk.  Upon completion, The function returns a list of 
+    filenames available on disk. 
+
+    Parameters
+    ----------
+    products : Table
+        A Table of products as returned by the mvm_id_filenames function. 
+
+    archive : Boolean, optional
+        Retain copies of the downloaded files in the astroquery created
+        sub-directories? Default is "False".
+
+    clobber : Boolean, optional
+        Download and Overwrite existing files? Default is "False".
+
+    verbose : bool
+        Default False.  If True, additional intermediate information is printed.
+
+    Returns
+    -------
+    local_files : list
+        List of filenames
+
+    Note: Code here cribbed from retrieve_obsevation in astroquery_utils module.
+    """
+
+    all_images = []
+    all_images = products['productFilename'].tolist()
+    if not clobber:
+        rows_to_remove = []
+        for row_idx, row in enumerate(products):
+            fname = row['productFilename']
+            if os.path.isfile(fname):
+                print(fname + " already exists. File download skipped.")
+                rows_to_remove.append(row_idx)
+        products.remove_rows(rows_to_remove)
+
```

Comment:
What happens if you remove all the rows at this point; namely, the user already had all the images they want locally and just wasn't sure or merely following a prescribed flow without understanding it?

Reply:
Added a check and log messages to let the user know the if files are downloading or all the files are already resident on disk.